provisioning/baremetal_secrets.go

```diff
@@ -236,9 +234,9 @@ func CreateOrUpdateTlsSecret(client coreclientv1.SecretsGetter, targetNamespace
 			Name:      tlsSecretName,
 			Namespace: targetNamespace,
 		},
-		StringData: map[string]string{
-			tlsCertificateKey: cert.certificate,
-			tlsPrivateKeyKey:  cert.privateKey,
+		Data: map[string][]byte{
```

Comment:
are you looking for `ApplySecret` from https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourceapply/core.go#L237 ?

Many resources have gotchas and that package was written by the original API authors to handle the gotchas.  You may want to consider grabbing a dep on it.